src/sema/expression.rs

```diff
@@ -781,28 +780,23 @@ impl Expression {
                         Type::Uint(address_bits)
                     };
 
-                    let expr = if *from_len > address_bits {
-                        Expression::Trunc {
+                    let expr = match from_len.cmp(&address_bits) {
+                        Ordering::Greater => Expression::Trunc {
                             loc: *loc,
                             to: address_to_int,
                             expr: Box::new(self.clone()),
-                        }
-                    } else if *from_len < address_bits {
-                        if from.is_signed_int() {
-                            Expression::ZeroExt {
-                                loc: *loc,
-                                to: address_to_int,
-                                expr: Box::new(self.clone()),
-                            }
-                        } else {
-                            Expression::SignExt {
-                                loc: *loc,
-                                to: address_to_int,
-                                expr: Box::new(self.clone()),
-                            }
-                        }
-                    } else {
-                        self.clone()
+                        },
+                        Ordering::Less if from.is_signed_int() => Expression::ZeroExt {
+                            loc: *loc,
+                            to: address_to_int,
+                            expr: Box::new(self.clone()),
+                        },
+                        Ordering::Less => Expression::SignExt {
+                            loc: *loc,
+                            to: address_to_int,
+                            expr: Box::new(self.clone()),
+                        },
+                        Ordering::Equal => self.clone(),
```

Comment:
I don't think this is necessarily easier to read, but to be fair I'm not opposed to it.